c7n/resources/account.py

```diff
@@ -424,6 +425,66 @@ def process(self, resources, event=None):
         return []
 
 
+@actions.register('set-password-policy')
+class SetAccountPasswordPolicy(BaseAction):
+    """Set an account's password policy.
+
+    This only changes the policy for the items provided.
+    If this is the first time setting a password policy and an item is not provided it will be
+    set to the defaults defined in the boto docs for IAM.Client.update_account_password_policy
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-account-password-policy
+                resource: account
+                filters:
+                  - not:
+                    - type: password-policy
+                      key: MinimumPasswordLength
+                      value: 10
+                      op: ge
+                actions:
+                    - type: set-password-policy
+                      policy:
+                        MinimumPasswordLength: 20
+    """
+    schema = type_schema(
+        'set-password-policy',
+        policy={
+            'type': 'object'
+        })
+    shape = 'UpdateAccountPasswordPolicyRequest'
+    service = 'iam'
+    permissions = ('iam:GetAccountPasswordPolicy', 'iam:UpdateAccountPasswordPolicy')
+
+    def validate(self):
+        return shape_validate(
+            self.data.get('policy', {}),
+            self.shape,
+            self.service)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        account = resources[0]
+        if account.get('c7n:password_policy'):
+            config = account['c7n:password_policy']
+        else:
+            try:
+                config = client.get_account_password_policy().get('PasswordPolicy')
+            except client.exceptions.NoSuchEntityException:
+                config = {}
+        for k, v in self.data.get('policy').items():
+            config[k] = v
+        config = shape_strip(
```

Comment:
I'm unclear why this is being done, your validating if the attribute is in the shape, but thats already part of validation.

Reply:
oh.. I see from previous this was about Expire, I would just strip off that key.. this use case isn't common enough for a helper imo.

Reply:
PasswordPolicyConfigured is being added in the password-policy filter so if that is cached I also need to remove it. Was thinking is a way to safely future proof it too, e.g. something new gets added later.

Reply:
generally unless something is being used in multiple places, then i'd consider the abstraction a tech debt, as it causes indirection to a reader, its better to refactor to abstractions then having them for one time usage.

Reply:
Alright, removed in ca5e3f3. I take this also as another cue to be more explicit of what is being done in the function. Thanks for the review!